azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -160,38 +198,142 @@ def __get_agent_family_manifests(self, goal_state):
         return agent_family_manifests[0]
 
     @staticmethod
-    def __get_requested_version(agent_family):
+    def _get_version_from_gs(agent_family):
         """
-        Get the requested version from agent family
-        Returns: Requested version if supported and available in the GS
-                 None if requested version missing or GA versioning not enabled
+        Get the version from agent family
+        Returns: version if supported and available in the GS
+                 None if version is missing
```

Comment:
Returning None makes the code harder to write/review/maintain.

For self-update we should not even be looking at the version in the goal state. If it is missing, we do not care. 

For RSM update, the version must not be missing, so you can raise an update exception here to report the error and skip the update, instead of returning None.

This is a consequence of my other comment below: that we are looking at the version before we have decided whether we are doing self-update or RSM update